packages/core/src/modules/credentials/CredentialsApi.ts

```diff
@@ -515,6 +525,87 @@ export class CredentialsApi<CPs extends CredentialProtocol[]> implements Credent
     return credentialRecord
   }
 
+  /**
+   * Revoke a credential by issuer
+   * associated with the credential record.
+   *
+   * @param credentialRecordId The id of the credential record for which to revoke the credential
+   * @returns Revoke credential notification message
+   *
+   */
+  public async revokeJsonLdCredential(options: RevokeCredentialOption): Promise<{ message: string }> {
```

Comment:
there is a lot of dupcliation in this file with the already present methods. Some suggestions:
- move all the credential status logic into a specific folder in the w3c module
- add the revocation related methods to the w3c credentails api, not the DIDcomm credentials api
- the only thing we should add it here is a `sendRevocationNotification` method (which we can extend to support non-anoncreds revocadion notification messages as well)
- making a POST to the credential status list url to update the status list seems implementation specific and may not work for all implementations